runtime/caml/signals.h

```diff
@@ -30,6 +30,19 @@
 extern "C" {
 #endif
 
+
+CAMLextern void caml_enter_blocking_section (void);
+CAMLextern void caml_leave_blocking_section (void);
+
+CAMLextern void caml_process_pending_events (void);
+CAMLextern value caml_process_pending_events_with_root (value extra_root);
```

Comment:
About "event": ok, a visit to the thesaurus allows me to propose "action" as a substitute.

I am rather unconvinced about the use of flags because:
- they are not going to be used programmatically,
- the flag would need to determine the return type. We could cheat, but this is bad.
On the contrary, I simply reuse the existing convention from `call_callback*_exn`, so the added learning effort for the programmer is minimal.

Reply:
The size of the runtime API keeps growing and having several subtly different variants of the same function does not help understanding/remembering it. Sure, naming patterns help, but flags help factorizing the API and the implementation, forces the user to think about the issue of exceptions, and makes the API more extensible. Note that you will probably use them programmatically to forward calls to these functions from one to another.

Actually, I could even propose you to factorize `caml_process_pending_events` and `caml_process_pending_gc` using another flag. The argument is the same : factorize subtly different functions (and thus make it clear that they serve similar purpose), and force the user to think about the value of the flags instead of calling the first functions that seems to fit its need.

Reply:
I have changed the semantics of caml_check_urgent_gc back, and one consequence is that there are a lot fewer variants. I would have liked to avoid the with_root variant and this is now the case given that there are no more uses of `process_pending_events_with_root` in the wild (the existence of such a function was a side-effect of making check_urgent_gc poll). caml_check_urgent_gc is no longer deprecated and keeps its original meaning.

In the end, the public interface has only two new functions: `caml_process_pending_actions{,_exn}`. Let me know if this reassures regarding the growing size of API. I plan to keep reusing the same convention in the future (the one from caml_callback{,_exn} and caml_startup{,_exn}).

Reply:
Good!